tests/unit/test_local_configuration.py

```diff
@@ -1,44 +1,42 @@
 import multiprocessing
 import os
 import sys
-import tempfile
 import unittest
 
+from needy.filesystem import TempDir
 from needy.local_configuration import LocalConfiguration
 
 
+def try_locking_local_config(path):
+    with LocalConfiguration(path, blocking=False) as config:
+        sys.exit(0 if config is not None else 1)
+
+
 class LocalConfigurationTest(unittest.TestCase):
     def test_locking(self):
-        file = tempfile.NamedTemporaryFile(delete=True)
-        try:
-            self.assertTrue(LocalConfigurationTest.try_access_from_another_process(file.name))
-            with LocalConfiguration(file.name) as config:
+        with TempDir() as temp_dir:
+            config_file = os.path.join(temp_dir, 'config')
+            self.assertTrue(LocalConfigurationTest.try_access_from_another_process(config_file))
+            with LocalConfiguration(config_file) as config:
                 self.assertIsNotNone(config)
-                self.assertFalse(LocalConfigurationTest.try_access_from_another_process(file.name))
-            self.assertTrue(LocalConfigurationTest.try_access_from_another_process(file.name))
-        finally:
-            file.close()
+                self.assertFalse(LocalConfigurationTest.try_access_from_another_process(config_file))
+            self.assertTrue(LocalConfigurationTest.try_access_from_another_process(config_file))
 
     def test_development_mode_persistence(self):
-        file = tempfile.NamedTemporaryFile(delete=True)
-        try:
-            with LocalConfiguration(file.name) as config:
+        with TempDir() as temp_dir:
+            config_file = os.path.join(temp_dir, 'config')
```

Comment:
Is there already a test to ensure the local config is created if it doesn't exist?